lib/explorer/data_frame.ex

```diff
@@ -1932,7 +1940,11 @@ defmodule Explorer.DataFrame do
           )
     end)
 
-    Shared.apply_impl(df, :take, [row_indices])
+    Shared.apply_impl(df, :slice, [row_indices])
+  end
+
+  def slice(df, %Range{} = range) do
+    slice(df, Enum.slice(0..(n_rows(df) - 1), range))
```

Comment:
I don't know if we can have an empty dataframe but just in case:

```suggestion
    slice(df, Enum.slice(0..(n_rows(df) - 1)//1, range))
```